safekeeper/src/safekeeper.rs

```diff
@@ -153,7 +155,7 @@ pub struct PeerInfo {
 impl PeerInfo {
     fn new() -> Self {
         Self {
-            s3_wal_lsn: Lsn(0),
+            backup_lsn: Lsn::INVALID,
```

Comment:
Just interested: are those `Lsn(0)` below valid or also invalid, but just not changed?
Might be a bit confusing to read and even the former is true (Lsn(0) and Lsn::Invalid are different by the meaning), then we'd rather express it with an enum instead?

Reply:
`Lsn(0)` is always invalid. Enum sounds nice, but it's not much convenient to use in the code.

Reply:
I would rewrite all of the LSN(0) as Lsn::INVALID _(and/or introduce Lsn::LOWEST_VALID where valid LSN smaller than anything Postgres Could write is semantically appropriate)_ but it will bloat the PR massively.
I'd say a dedicated pr for such change is a better idea.